centralized-sampling-tests/integration-tests/src/main/java/com.amazon.tests/CentralizedSamplingIntegrationTests.java

```diff
@@ -0,0 +1,313 @@
+package com.amazon.tests;
+
+import java.io.*;
+import java.lang.Math;
+import java.util.concurrent.TimeUnit;
+import okhttp3.*;
+import org.springframework.context.annotation.Bean;
+
+/** File that runs all of the integrations tests and creates/deletes sample rules */
+public class CentralizedSamplingIntegrationTests {
+
+  /**
+   * Set up new httpClient
+   *
+   * @return OkhttpClient
+   */
+  @Bean
+  public static Call.Factory httpClient() {
+    return new OkHttpClient();
+  }
+
+  /**
+   * Main function that runs the three tests for Centralized Sampling SampleRulesTests - tests rules
+   * filter and sample the correct number of targets ReservoirTests - tests that reservoir works
+   * correctly with Sampling Rules PriorityTests - test Priority of sampling rules that higher
+   * priority rules get sampled first
+   *
+   * @param args
+   * @throws IOException - throws if calls to xray backend fail
+   * @throws InterruptedException - throws if tests fail
+   */
+  public static void main(String[] args) throws IOException, InterruptedException {
+    sampleRulesTests();
```

Comment:
Wondering what is major difference in 3 tests? They all make rule, go though cases, verify the response with retry.

Reply:
I documented this in sampleRules, but probably worth documenting here. The difference is in how the rules are set and when the calls are made. sampleRulesTests creates 1 rule, calls all tests cases and repeat. Reservoir tests creates 1 rule, waits x amount of time because of the need to wait for the reservoir to fill and repeat. Priority test creates all rules then calls all test cases to test priority.